common/src/test/java/org/apache/sedona/common/raster/RasterEditorsTest.java

```diff
@@ -134,6 +134,61 @@ public void testResample() throws FactoryException, TransformException {
         }
     }
 
+    @Test
+    public void testResampleNoDataValueHandled() throws FactoryException, TransformException {
+        double[] values1 = {1,99,3,4,99,6,7,99,9,10,99,12};
+        double[] values2 = {10,11,-9999,13,14,15,-9999,17,18,19,-9999,21};
+        GridCoverage2D raster = RasterConstructors.makeEmptyRaster(1, "d", 4, 3, 0, 0, 2, -2, 0, 0, 0);
+        raster = MapAlgebra.addBandFromArray(raster, values1, 1, 99.0);
+        raster = MapAlgebra.addBandFromArray(raster, values2, 2, -9999.0);
+
+        GridCoverage2D newRaster = RasterEditors.resample(raster, 6, 5, 1, -1, false, "bilinear");
```

Comment:
Can you add a test to downsample the raster? E.g., the original raster is 4, 3, resample it to 2, 2?